docs/troubleshooting/agent.md

```diff
@@ -0,0 +1,109 @@
+things could be happening:
+
+* The connection between the switch and the control node is down or misconfigured.
+* The agent isn't running on the switches
+* The agent is stuck trying to apply a configuration
+* The time is not synchronized between the controller and the switches 
+
+#### Switch and Controller Connection
+
+The control node serves DHCP to the switches as part of normal operation and
+replies to ONIE requests for NOS install. The controller expects layer 2
+reachability and to be the only DHCP server in the LAN. This architecture
+can take many forms but it is most often a dedicated switch or VLAN.
+Ensure that the link on the control node is in the `up` state. Often problems
+in this area are related to the configuration of a private VLAN, or cabling.
+
+#### Starting and Stopping the Agent
+
+The switch agent is a service that is controlled via `systemctl` commands.
+The agent is started automatically at switch bootup time, to stop the agent:
+```bash
+sudo systemctl stop hedgehog-agent.service
+```
+similarly to start the agent:
+```bash
+sudo systemctl start hedgehog-agent.service
+```
+
+Stopping the agent should only be done as part of debugging, because the agent will
+reapply the configuration on a regular interval during normal operation.
+
+#### Restoring Agent State
+
+In rare circumstances, the agent can become stuck trying to apply a
+configuration. Before attempting these steps reach out to Hedgehog as this is
+likely a bug. If an agent is stuck, the configuration can be
+generated on the control node using `kubectl`, then manually moved to the
+switch. The example below uses the name of a switch for illustrative purposes.
+
+1. on control node: `kubectl get -o yaml agents/switch-name > agent-config.yaml`
+1. on control node: `scp agent-config.yaml admin@leaf-04:/tmp/agent-config.yaml`
+1. on control node: `ssh admin@leaf-04`
+1. on switch: `sudo mv /tmp/agent-config.yaml /etc/sonic/hedgehog/agent-config.yaml`
+
+#### Time Synchronization
+
+The control node is configured to use public NTP servers from Google or
+Cloudflare. The control node will also act as an NTP server for the switches.
+
+To check the status of the NTP service on the control node, use the `systemctl`
+command:
+
+```bash
+systemctl status chronyd.service
+```
+
+Confirm that the switches are using the control node as their NTP source:
+
+```bash
+ntpstat
+```
+
+The Hedgehog agent sets the NTP configuration on the switch.
+
+## Applied, AppliedG, and CurrentG
+
+Each configuration on the switches is tracked as a generation. The `APPLIEDG`
+column indicates the generation that is the running configuration on the switch.
+After a change is made, the `CURRENTG` column will be higher than the `APPLIEDG`
+The switch agent will apply the changes and the `CURRENTG` column will equal
+the `APPLIEDG` column. The `APPLIED` column displays the amount of time that
+has passed since the last configuration was applied.
+
+## ROCE, ECMPQPN
```

Comment:
@Frostman, is this correct? Is the CURROCE column where we indicate if a switch is rebooting to ROCE mode?

Reply:
```
// +kubebuilder:printcolumn:name="RoCE",type=boolean,JSONPath=`.spec.switch.roce`,priority=3
// +kubebuilder:printcolumn:name="CurrRoCE",type=boolean,JSONPath=`.status.state.roce`,priority=3
// +kubebuilder:printcolumn:name="ECMPQPN",type=boolean,JSONPath=`.spec.switch.ecmp.roceQPN`,priority=3
```

- `RoCE` is desired
-  `CurrRoCE` is current or last reported by the switch
-  `ECMPQPN` is desired